#! /bin/sh /usr/share/dpatch/dpatch-run
## 12_changeset_16483.dpatch by Mario Limonciello <superm1@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix DVD menu handling. See MythTV #4897 for more info.

@DPATCH@
diff -urNad mythtv-0.21.0~/libs/libmythtv/NuppelVideoPlayer.cpp mythtv-0.21.0/libs/libmythtv/NuppelVideoPlayer.cpp
--- mythtv-0.21.0~/libs/libmythtv/NuppelVideoPlayer.cpp	2008-03-07 18:07:31.000000000 -0600
+++ mythtv-0.21.0/libs/libmythtv/NuppelVideoPlayer.cpp	2008-03-09 20:19:51.000000000 -0500
@@ -996,8 +996,11 @@
         return;
     }
 
-    video_dim    = QSize((width + 15) & ~0xf, (height + 15) & ~0xf);
-    video_disp_dim=QSize(width, height);
+    if ((width > 0) && (height > 0))
+    {
+        video_dim      = QSize((width + 15) & ~0xf, (height + 15) & ~0xf);
+        video_disp_dim = QSize(width, height);
+    }
     video_aspect = (aspect > 0.0f) ? aspect : video_aspect;
     keyframedist = (keyframedistance > 0) ? keyframedistance : keyframedist;
 
